src/common/walFilter/versions/recordProcessGPDB6.c

```diff
@@ -0,0 +1,474 @@
+getSpgist(const XLogRecord *record)
+{
+    const uint8 info = (uint8) (record->xl_info & ~XLR_INFO_MASK);
+
+    switch (info)
+    {
+        case XLOG_SPGIST_CREATE_INDEX:
+        case XLOG_SPGIST_ADD_LEAF:
+        case XLOG_SPGIST_MOVE_LEAFS:
+        case XLOG_SPGIST_ADD_NODE:
+        case XLOG_SPGIST_SPLIT_TUPLE:
+        case XLOG_SPGIST_PICKSPLIT:
+        case XLOG_SPGIST_VACUUM_LEAF:
+        case XLOG_SPGIST_VACUUM_ROOT:
+        case XLOG_SPGIST_VACUUM_REDIRECT:
+            return (RelFileNode *) XLogRecGetData(record);
+    }
+    THROW_FMT(FormatError, "SPGIST UNKNOWN: %d", info);
+}
+
+// Get RelFileNode from Bitmap record.
+static RelFileNode *
+getBitmap(const XLogRecord *record)
+{
+    const uint8 info = (uint8) (record->xl_info & ~XLR_INFO_MASK);
+    switch (info)
+    {
+        case XLOG_BITMAP_INSERT_LOVITEM:
+        case XLOG_BITMAP_INSERT_META:
+        case XLOG_BITMAP_INSERT_BITMAP_LASTWORDS:
+        case XLOG_BITMAP_INSERT_WORDS:
+        case XLOG_BITMAP_UPDATEWORD:
+        case XLOG_BITMAP_UPDATEWORDS:
+            return (RelFileNode *) XLogRecGetData(record);
+    }
+    THROW_FMT(FormatError, "Bitmap UNKNOWN: %d", info);
+}
+
+// Get RelFileNode from Appendonly record.
+static RelFileNode *
+getAppendonly(const XLogRecord *record)
+{
+    const uint8 info = (uint8) (record->xl_info & ~XLR_INFO_MASK);
+    switch (info)
+    {
+        case XLOG_APPENDONLY_INSERT:
+        case XLOG_APPENDONLY_TRUNCATE:
+            return (RelFileNode *) XLogRecGetData(record);
+    }
+    THROW_FMT(FormatError, "Appendonly UNKNOWN: %d", info);
+}
+
+FN_EXTERN const RelFileNode *
+getRelFileNodeGPDB6(const XLogRecord *record)
+{
+    switch (record->xl_rmid)
+    {
+        case RM_XLOG_ID:
+            return getXlog(record);
+
+        case RM_SMGR_ID:
+            return getStorage(record);
+
+        case RM_HEAP2_ID:
+            return getHeap2(record);
+
+        case RM_HEAP_ID:
+            return getHeap(record);
+
+        case RM_BTREE_ID:
+            return getBtree(record);
+
+        case RM_GIN_ID:
+            return getGin(record);
+
+        case RM_GIST_ID:
+            return getGist(record);
+
+        case RM_SEQ_ID:
+            return getSeq(record);
+
+        case RM_SPGIST_ID:
+            return getSpgist(record);
+
+        case RM_BITMAP_ID:
+            return getBitmap(record);
+
+        case RM_APPEND_ONLY_ID:
+            return getAppendonly(record);
+
+        // Records of these types do not contain a RelFileNode.
+        case RM_XACT_ID:
+        case RM_CLOG_ID:
+        case RM_DBASE_ID:
+        case RM_TBLSPC_ID:
+        case RM_MULTIXACT_ID:
+        case RM_RELMAP_ID:
+        case RM_STANDBY_ID:
+        case RM_DISTRIBUTEDLOG_ID:
+            // skip
+            return NULL;
+
+        case RM_HASH_ID:
+            THROW(FormatError, "Not supported in GPDB6. Shouldn't be here");
+
+        default:
+            THROW(FormatError, "Unknown resource manager");
+    }
```

Comment:
```suggestion
    }
    THROW(FormatError, "Unknown resource manager");
```
because `record->xl_rmid` is not enum

Reply:
done